doc/stdenv/cross-compilation.chapter.md

```diff
@@ -0,0 +1,167 @@
+
+Finally, if the depending package is a compiler or other machine-code-producing tool, it might need dependencies that run at "emit time". This is for compilers that (regrettably) insist on being built together with their source langauges' standard libraries. Assuming build != host != target, a run-time dependency of the standard library cannot be run at the compiler's build time or run time, but only at the run time of code emitted by the compiler.
+
+Putting this all together, that means we have dependencies in the form "host → target", in at most the following six combinations:
+
+
+#### Possible dependency types
+| Dependency's host platform | Dependency's target platform |
+| --                         | --                           |
+| build                      | build                        |
+| build                      | host                         |
+| build                      | target                       |
+| host                       | host                         |
+| host                       | target                       |
+| target                     | target                       |
+
+
+Some examples will make this table clearer. Suppose there's some package that is being built with a `(build, host, target)` platform triple of `(foo, bar, baz)`. If it has a build-time library dependency, that would be a "host → build" dependency with a triple of `(foo, foo, *)` (the target platform is irrelevant). If it needs a compiler to be built, that would be a "build → host" dependency with a triple of `(foo, foo, *)` (the target platform is irrelevant). That compiler, would be built with another compiler, also "build → host" dependency, with a triple of `(foo, foo, foo)`.
+
+### Cross packaging cookbook {#ssec-cross-cookbook}
+
+Some frequently encountered problems when packaging for cross-compilation should be answered here. Ideally, the information above is exhaustive, so this section cannot provide any new information, but it is ludicrous and cruel to expect everyone to spend effort working through the interaction of many features just to figure out the same answer to the same common problem. Feel free to add to this list!
+
+**Q:** What if my package's build system needs to build a C program to be run under the build environment?
+```
+depsBuildBuild = [ buildPackages.stdenv.cc ];
+```
+Add it to your `mkDerivation` invocation.
+
+**Q:** My package fails to find `ar`.
+
+**A:** Many packages assume that an unprefixed `ar` is available, but Nix doesn't provide one. It only provides a prefixed one, just as it only does for all the other binutils programs. It may be necessary to patch the package to fix the build system to use a prefixed `ar`.
+
+**Q:** My package's testsuite needs to run host platform code.
```

Comment:
This last question is missing an answer.

Reply:
It's silly that some of the questions aren't actually questions, but we can fix that with later content revisions.